api/types/server.go

```diff
@@ -392,12 +394,39 @@ func (s *ServerV2) CheckAndSetDefaults() error {
 	// defaults.ServerAnnounceTTL).
 	s.setStaticFields()
 	if err := s.Metadata.CheckAndSetDefaults(); err != nil {
-		return trace.Wrap(err)
+		// if the server is a registered OpenSSH node, allow the name
```

Comment:
Should be pretty easy to add some test coverage for this new logic.

Reply:
Done